src/pe/mod.rs

```diff
@@ -92,6 +104,47 @@ impl<'a> PE<'a> {
         let mut certificates = Default::default();
         let mut is_64 = false;
         if let Some(optional_header) = header.optional_header {
+            #[cfg(feature = "pe_authenticode")]
+            // Sections we are assembling through the parsing, eventually, it will be passed
+            // to the authenticode_sections attribute of `PE`.
+            let mut excluded_sections = {
+                let (checksum, datadir_entry_certtable) =
+                    match optional_header.standard_fields.magic {
+                        optional_header::MAGIC_32 => {
+                            let standard_field_offset =
+                                optional_header_offset + optional_header::SIZEOF_STANDARD_FIELDS_32;
+                            let checksum_field_offset = standard_field_offset
+                                + optional_header::OFFSET_WINDOWS_FIELDS_32_CHECKSUM;
+                            (
+                                checksum_field_offset..checksum_field_offset + 4,
+                                optional_header_offset + 128..optional_header_offset + 136,
+                            )
+                        }
+                        optional_header::MAGIC_64 => {
+                            let standard_field_offset =
+                                optional_header_offset + optional_header::SIZEOF_STANDARD_FIELDS_64;
+                            let checksum_field_offset = standard_field_offset
+                                + optional_header::OFFSET_WINDOWS_FIELDS_64_CHECKSUM;
+
+                            (
+                                checksum_field_offset..checksum_field_offset + 4,
+                                optional_header_offset + 144..optional_header_offset + 152,
+                            )
+                        }
+                        magic => {
+                            return Err(error::Error::Malformed(format!(
+                                "Unsupported header magic ({magic:#x})"
+                            )))
+                        }
+                    };
+
+                authenticode::ExcludedAuthenticodeSections {
```

Comment:
i think you should extract this into section into a `pub(super) fn new()` for `authenticode::ExcludedAuthenticodeSections`